patroni/ctl.py

```diff
@@ -1404,6 +1416,151 @@ def _do_failover_or_switchover(action: str, cluster_name: str, group: Optional[i
     output_members(cluster, cluster_name, group=group)
 
 
+def _do_site_switchover(cluster_name: str, group: Optional[int],
+                        switchover_leader: Optional[str], candidate: Optional[str],
+                        force: bool, scheduled: Optional[str] = None) -> None:
+    """Perform a site switchover operation in the cluster.
+
+    Informational messages are printed in the console during the operation, as well as the list of members before and
+    after the operation, so the user can follow the operation status.
+
+    .. note::
+        If not able to perform the operation through the REST API, write directly to the DCS as a fall back.
+
+    :param cluster_name: name of the Patroni cluster.
+    :param group: filter Citus group within we should perform a failover or switchover. If ``None``, user will be
+        prompted for filling it -- unless *force* is ``True``, in which case an exception is raised.
+    :param switchover_leader: name of the leader site passed as switchover option.
+    :param candidate: name of a standby site to be promoted.
+    :param force: perform the switchover without asking for confirmations.
+    :param scheduled: timestamp when the switchover should be scheduled to occur. If ``now`` perform immediately.
+
+    :raises:
+        :class:`PatroniCtlException`: if:
+            * Patroni is running on a Citus cluster, but no *group* was specified; or
+            * a switchover was requested but the cluster has no leader site (or multisite is not active); or
+            * *switchover_leader* does not match the current leader site of the cluster; or
+            * cluster has no candidate sites available for the operation; or
+            * current leader and *candidate* are the same; or
+            * *candidate* is not a site of the cluster; or
+            * trying to schedule a switchover in a cluster that is in maintenance mode; or
+            * user aborts the operation.
+    """
+    dcs = get_dcs(cluster_name, group)
+    cluster = dcs.get_cluster()
+    click.echo('Current cluster topology')
+    output_members(cluster, cluster_name, group=group)
+
+    if is_citus_cluster() and group is None:
```

Comment:
Citus will be quite unhappy if some nodes don't have a local leader. Correct fix is probably to have a single point that determines which site is active for the whole cluster. But that is a larger job. I think for now we should just have an error saying that citus and multisite together is unsupported.